core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js

```diff
@@ -231,9 +231,11 @@ var TopicsAndSkillsDashboardPage = function() {
   this.filterSkillsByStatus = async function(status) {
     await action.click(
       'Skill Dashboard status filter', skillStatusFilterDropdown);
-    await (
-      await browser.driver.switchTo().activeElement()
-    ).sendKeys(status + '\n');
+    var filterSkillStatus = skillStatusFilterDropdown.element(
+      by.css('.protractor-test-select-skill-status-dropdown'));
+
+    await action.sendKeys(
+      'Skill Dashboard status filter', filterSkillStatus, status + '\n');
```

Comment:
The `e2e_non_prod_miscellaneous` tests failed 3 times in a row with with this error:

```
1) Contributor dashboard page should allow user to switch to translate text tab
  Message:
    Failed: Skill Dashboard status filter is not visible.
    Wait timed out after 10011ms
  Stack:
    TimeoutError: Skill Dashboard status filter is not visible.
    Wait timed out after 10011ms
        at /home/circleci/oppia/node_modules/selenium-webdriver/lib/promise.js:2201:17
        at runMicrotasks (<anonymous>)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
    From asynchronous test: 
    Error
        at Suite.<anonymous> (/home/circleci/oppia/core/tests/protractor_desktop/contributorDashboard.js:53:3)
        at Object.<anonymous> (/home/circleci/oppia/core/tests/protractor_desktop/contributorDashboard.js:34:1)
        at Module._compile (internal/modules/cjs/loader.js:1063:30)
        at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
        at Module.load (internal/modules/cjs/loader.js:928:32)
        at Function.Module._load (internal/modules/cjs/loader.js:769:14)
```

Here's a link to the failing build: https://app.circleci.com/pipelines/github/oppia/oppia/18584/workflows/47dc5e2c-a1ba-471e-8ecd-1a980a1a0120/jobs/105083

Are you sure your `filterSkillStatus` element is correct?

Reply:
Hi @U8NWXD,
I'm not sure why this is happening I mean when I'm checking it locally it's working fine.

Reply:
So the test passes locally with your changes?

Reply:
@U8NWXD  No, I mean when I access `filterSkillStatus` locally in my server it was working fine. I'm trying to figure out why this is happening but not getting anything.
Thanks

Reply:
It might help to run the test locally and watch what happens. You can also use the debugger to inspect the DOM at each step. Instructions for using the debugger are in the [wiki](https://github.com/oppia/oppia/wiki/End-to-End-Tests#steps-to-use-the-debugger-tool). You can also comment-out unrelated test code to focus on your issue

Reply:
Hi @U8NWXD, Sorry for this much late update,
When I'm running tests locally, tests are failing in recognizing the usernames. It shows an error on the screen like `Invalid username: translator` and it fails for several usernames. Even after merging with upstream develop.
Also, I got to know that my CSS element is not correct when I'm sending keys, So I again searched for that, which element is active when we hit the dropdown but no specific class highlighted at that particular moment. According to me, it should be `.select2-selection__rendered`, will you please take a look and confirm because my tests are again failing.

I'm facing a strange situation, I mean if I have to start the development server or I have to run tests locally, then each time I have to restart my laptop, otherwise, the development server won't start even running the start script several times, do you know what's causing this issue and should I report this on gitter channel?
Thanks

Reply:
@lkbhitesh07 this might be because some of the servers started by the e2e tests haven't shut down properly. Elasticsearch commonly does this. You can search for it like this on unix/linux systems:

```console
px -ax | grep elasticsearch
```

The `grep` process will always show up, but if you see any others, you can `kill` them

Reply:
I think you found the right element. Since this doesn't seem to be working, you could also try using the `action.select2` function:

https://github.com/oppia/oppia/blob/730e3eeaf7063a4a07b4848ea2e2bd13fd9a6874/core/tests/protractor_utils/action.js#L61-L69

Reply:
Hi @U8NWXD, this error is resolved but getting another error, PTAL, Also I'm committing those changes in case you want to see full errors.
Thanks
```
1) Contributor dashboard page should allow user to switch to translate text tab
  Message:
    Failed: javascript error: Cannot read property 'indexOf' of null
      (Session info: chrome=88.0.4324.150)
      (Driver info: chromedriver=88.0.4324.96 (68dba2d8a0b149a1d3afac56fa74648032bcf46b-refs/branch-heads/4324@{#1784}),platform=Linux 4.15.0-135-generic x86_64)
  Stack:
    JavascriptError: javascript error: Cannot read property 'indexOf' of null
      (Session info: chrome=88.0.4324.150)
      (Driver info: chromedriver=88.0.4324.96 (68dba2d8a0b149a1d3afac56fa74648032bcf46b-refs/branch-heads/4324@{#1784}),platform=Linux 4.15.0-135-generic x86_64)
        at Object.checkLegacyResponse (/home/ht/opensource/gsoc/oppia/node_modules/selenium-webdriver/lib/error.js:546:15)
        at parseHttpResponse (/home/ht/opensource/gsoc/oppia/node_modules/selenium-webdriver/lib/http.js:509:13)
        at /home/ht/opensource/gsoc/oppia/node_modules/selenium-webdriver/lib/http.js:441:30
        at runMicrotasks (<anonymous>)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
    From asynchronous test: 
    Error
        at Suite.<anonymous> (/home/ht/opensource/gsoc/oppia/core/tests/protractor_desktop/contributorDashboard.js:53:3)
        at Object.<anonymous> (/home/ht/opensource/gsoc/oppia/core/tests/protractor_desktop/contributorDashboard.js:34:1)
        at Module._compile (internal/modules/cjs/loader.js:1063:30)
        at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
        at Module.load (internal/modules/cjs/loader.js:928:32)
        at Function.Module._load (internal/modules/cjs/loader.js:769:14)
  Message:
    Expected $.length = 1 to equal 0.
    Expected $[0] = Entry({ level: SEVERE, message: 'https://www.googletagmanager.com/gtag/js - Failed to load resource: net::ERR_CONNECTION_CLOSED', timestamp: 1614097357613, type: '' }) to equal undefined.
  Stack:
    Error: Failed expectation
        at Object.checkForConsoleErrors (/home/ht/opensource/gsoc/oppia/core/tests/protractor_utils/general.js:64:23)
        at runMicrotasks (<anonymous>)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
        at UserContext.<anonymous> (/home/ht/opensource/gsoc/oppia/core/tests/protractor_desktop/contributorDashboard.js:224:5)
```

Reply:
For the `portserver.socket` error, you can just delete that file. It's supposed to get cleaned up automatically by the server, but sometimes that cleanup code doesn't run if the server doesn't exit cleanly

Reply:
Unfortunately the stack traces don't pinpoint the failing line, but they do tell us that this is happening in the test at `protractor_desktop/contributorDashboard.js:53`. Unfortunately, you might have to comment out code until you identify the line where this is coming from. This will be the line that you can comment out to stop the error. I usually do this as a binary search where I first comment out the bottom half of the test. If the error still happens, then I comment out the bottom half of the uncommented section. If the error stops happening, then I uncomment the top half of the commented section, etc.

Reply:
Hi @U8NWXD , I don't know what but whenever I run my tests locally it just gets stuck, I tried to follow the guide you mentioned above but nothing seems to work and got overwhelmed, I have spent way much time in this org but wasn't able to make contributions. What should I do now, should I consider reinstalling the setup?

Reply:
Can you open an issue to describe what's going wrong when you run the tests? Please include screenshots and/or console output so that we can help you debug. I think an issue will be a better medium since this comment thread is getting pretty long at this point. Once you open it, you can at-mention and assign me so I get notified